formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -197,21 +211,19 @@ internal class JsonReader(private val source: String) {
             TC_END_LIST -> "end of the array ']'"
             else -> "valid token" // should never happen
         }
-        val s = if (currentPosition == source.length) "EOF" else source[currentPosition - 1].toString()
+        val s = if (currentPosition == source.length || currentPosition == 0) "EOF" else source[currentPosition - 1].toString()
```

Comment:
You may use `currentPosition <= 0`, just to make sure we've avoided arithmetic errors